src/ZoneServer/Buffs/Handlers/Swordsman/Barbarian/Frenzy_Buff.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using Melia.Shared.Game.Const;
+using Melia.Zone.Buffs.Base;
+using Melia.Zone.Network;
+using Melia.Zone.Skills;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.World.Actors;
+using Yggdrasil.Logging;
+
+namespace Melia.Zone.Buffs.Handlers.Swordsman.Barbarian
+{
+	/// <summary>
+	/// Handler for the Frenzy Buff, which increases attack speed
+	/// damage dealt, and damage taken as long as you keep
+	/// attacking the same target.
+	/// </summary>
+	/// <remarks>
+	/// Note that this buff stores the character you have frenzy
```

Comment:
The way it works is right, but I'm not sure if it's possible to see how the official is storing it.  You could probably also make a target the numarg1 or something.

There's also a buff for Cataphract which does something like this as well.

Reply:
I didn't mean the behavior in general, but the specific implementation^^ You would be able to see this in the buff packet, which contains the handles of the target and the caster.

Reply:
I suppose we could Zenmya it, though I have no real guesses as to whether or not they'd do it this way.  The main appeal of this implementation is it saves having to do a lookup using the handle every time we need the other entity.

Reply:
> it saves having to do a lookup
Just for reference, where do you need a lookup? You don't even use the caster, aside from checking if it's still the same, right? Which could also be accomplished by comparing handles. But while this implementation is a bit difficult to wrap your head around, I don't mind it if it makes the most sense.

Reply:
You're right, we could theoretically store the handle as numarg1 and use that.  The only issue is that numarg1 is a float and handle is an int.  The smallest int that can't be stored as a float is 16777217.  Do you think there's every any chance that the handles could reach this range?  For any target that had a handle that couldn't be stored as a float, it'd be impossible to build frenzy against them.

Reply:
You could also use buff or character variables, you don't need the num args. Actually, we should avoid using the buff args to save arbitrary information, because the arguments are sent to the client. They serve a purpose outside of us communicating information to the buff handlers.

Looking at the code for Frenzy again, is it actually correct that it starts on the first attack you make? The buff starts the moment you make a normal attack, and it immediately adds a 2% multiplier. Effectively this means Frenzy will always be on and you will always get a 2% bonus, no matter what. If that's not how it should go, if Frenzy should only activate on the second hit, you would actually need a different strategy, like storing the target in a scripting variable.

All that being said, if the current version works, we might not need to change it, unless you think it could be done better with relatively little efford.

Reply:
Actually, it would be very simple to make Frenzy's damage bonus not apply to the hit that builds it, all we'd have to do is reverse the order of the call to add the buff and apply the damage multiplier in Barbarian_Frenzy.cs.  If the buff adding was after the damage multiplier the damage bonus wouldn't apply to the hit that builds it.  That said, I feel like that's a bit more confusing and I suspect the buff is added first.

It's also important to note that although Frenzy only builds up with normal attacks, its damage bonus applies to skills as well.  The primary idea behind the skill is to build it up with a few normal attacks and then unleash a powerful skill hit with it.